mono/mini/mini-exceptions.c

```diff
@@ -1512,20 +1512,24 @@ mono_get_portable_ip (intptr_t in_ip, intptr_t *out_ip, gint32 *out_offset, cons
 	// Note: it's not safe for us to be interrupted while inside of dl_addr, because if we
 	// try to call dl_addr while interrupted while inside the lock, we will try to take a
 	// non-recursive lock twice on this thread, and will deadlock.
-	Dl_info info;
-	gboolean success = dladdr ((void*)in_ip, &info);
+	char *sname = NULL, *fname = NULL;
+	void *saddr = NULL, *fbase = NULL;
+	gboolean success = g_module_address ((void*)in_ip, &fname, &fbase, &sname, &saddr);
```

Comment:
So the fact that `g_module_address` dynamically allocates makes it unsuitable to be used right here.  `mono_get_portable_ip` is called by the crash reporter which runs in a signal handler (and also might be called when the malloc heap is corrupted) so it can't allocate.